games/static/games/proto.js

```diff
@@ -4,31 +4,56 @@ class Game {
     this.setup = setup;
     this.grid = document.querySelector(".grid");
     this.startButton = document.querySelector("#start");
+    this.restartButton = document.querySelector("#restart");
     this.isRunning = false
   }
   start() {
-    let alienMovementId = 0;
+    let gameId = 0;
+    let cleanId = 0;
+    let start
+    let restart
     let board = new Board(this.grid, BOARD_WIDTH, this.setup);
-    this.startButton.addEventListener("mousedown", () => {
-      if (alienMovementId) {
-        clearInterval(alienMovementId);
-        alienMovementId = null;
-        this.isRunning = false
+    this.startButton.addEventListener("mousedown", start = () => {
+      if (gameId) {
+        clearInterval(gameId);
+        gameId = null;
+        this.gamePause();
       } else {
-        this.isRunning = true
-        alienMovementId = setInterval(() => {
-          if (board.alienLocations.length === 0){
-            clearInterval(alienMovementId);
-            this.isRunning = false
-          } 
-          board.moveAliens();
-        }, 500);
+          this.gameRun();
+          gameId = setInterval(() => {
+            if (board.noAliensLeft()){
+              this.grid.innerHTML = '';
```

Comment:
mmm different levels of abstraction
this is too low level
move to its own function
something like board.annonceWinner